lib/dialogs/close-window-confirmation/index.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import actions from '../../state/actions';
+import UnsynchronizedConfirmation from '../unsynchronized';
+
+import type * as S from '../../state';
+
+type DispatchProps = {
+  action: () => any;
+};
+
+type Props = DispatchProps;
+
+const CloseWindowConfirmation = ({ action }: Props) => (
+  <UnsynchronizedConfirmation
+    description="Closing the app with unsynchronized notes could cause data loss."
+    unsafeAction="Close window"
+    safeAction="Safely close window"
+    action={action}
+  />
+);
+
+const mapDispatchToProps: S.MapDispatch<DispatchProps> = {
+  action: actions.electron.reallyCloseWindow,
```

Comment:
sorry for the duplicate comment - adding here since the last one might have been lost during the other code update.

this component, the `CloseWindowConfirmation`, isn't asking for some abstract `action` - it specifically wants to close the window and so why not maintain that honesty in the prop names. dialogs are already so confusing in this app and I don't want to see us keep the abstraction bleeding by exposing internals up the React ladder.

as a default-wrapped component nobody else should ever know but I don't think we named this `action` because that's what we felt was the component API of `CloseWindowConfirmation` - that would ring closer to what the `LogoutConfirmation` dialog previously used - in this case, `reallyCloseWindow`.

Reply:
resolved in https://github.com/Automattic/simplenote-electron/pull/2277/commits/094570fc5044611800546774d72d58d716113af6